model/relation/MediaRelationCollection.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ *
+ */
+
+namespace oat\taoMediaManager\model\relation;
+
+use ArrayIterator;
+use IteratorAggregate;
+use JsonSerializable;
+
+class MediaRelationCollection implements IteratorAggregate, JsonSerializable
+{
+    private $mediaRelations = [];
+
+    public function __construct(?array $mediaRelations = [])
```

Comment:
- [ ] We should have strong type hint here:

```suggestion
    public function __construct(MediaRelation ...$mediaRelations)
```

Reply:
Be doing like that I cannot create from array anymore

Reply:
The casting is done by the setter

Reply:
You can do by:

```
$collection = new MediaRelationCollection(...$myArray);
```